bin/src/Flags.purs

```diff
@@ -203,6 +203,15 @@ backendArgs =
           <> O.metavar "ARGS"
       )
 
+esbuildArgs :: Parser (List String)
+esbuildArgs =
+  O.many $
+    O.strOption
+      ( O.long "esbuild-args"
+          <> O.help "Arguments to pass to ESBuild"
+          <> O.metavar "ARGS"
+      )
+
```

Comment:
@f-f Thoughts on the flag name? 
Extra purs args are under --purs-args. 

But the workflow config presently has a generic field `extra_args` instead of `esbuild_args` for the field.

Reply:
Let's go with a more generic `bundler-args`

Reply:
Doesn't that imply some false degree of generality?

As soon as a spago user starts relying on this flag to pass extra args, they are relying on the esbuild API. Hypothetically, if spago ever switches from bundling with esbuild to something else, then this user's use case is broken. We might as well name the flag after the underlying tool for clarity. Given we can't actually safely change the bundler without breaking assumptions. 

 `--purs-args` also sets a bit of a precendent, vs. calling it `--compiler-args`.

Reply:
I'm thinking of alternate backends when I propose this: I'd like to make `spago bundle` call out to `$BACKEND bundle` if a backend is specified, in which case a esbuild-specific flag is not truthful anymore.

We had a similar situation with the old spago, where we first had `node-args`, then we added `backend-args` and the two did the same thing and it was confusing 😄